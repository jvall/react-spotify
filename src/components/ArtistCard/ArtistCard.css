.ArtistCard {
    width: 200px;
    margin: 10px;
    display: flex;
    flex-direction: column;
}

.ArtistCard img {
    width: 100%;
    height: 100px;
    object-fit: cover;
}

.name {
    font-weight: bold;
}