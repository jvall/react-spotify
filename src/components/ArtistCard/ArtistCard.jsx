import React from 'react';
import PropTypes from 'prop-types';
import './ArtistCard.css';

const ArtistCard = ({artist, likeArtist, unlikeArtist, likedArtists}) => {
    let handleClick;
    let buttonText = '';
    if(likedArtists.filter((likedArtist) => likedArtist.id === artist.id).length > 0) {
        handleClick = unlikeArtist;
        buttonText = 'Unlike';
    }
    else {
        handleClick = likeArtist;
        buttonText = 'Like';
    }
    return (
        <div className="ArtistCard">
            <img
                src={(artist.images && artist.images[0])
                ? artist.images[0].url
                : "http://placehold.it/200x200"}/>
            <div className="name">{artist.name}</div>
            <div className="genres">{artist
                    .genres
                    .join(', ')}</div>
            <button onClick={() => handleClick(artist)}>{buttonText}</button>
        </div>
    );
};

ArtistCard.propTypes = {
    artist: PropTypes.object.isRequired,
    likeArtist: PropTypes.func.isRequired,
    unlikeArtist: PropTypes.func.isRequired,
    likedArtists: PropTypes.array.isRequired
}

export default ArtistCard;

