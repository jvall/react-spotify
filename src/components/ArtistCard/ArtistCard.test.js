import React from 'react';
import { shallow } from 'enzyme';
import ArtistCard from './ArtistCard.jsx';
import renderer from 'react-test-renderer';

describe('<ArtistCard />', () => {

    let artist ={
        'id': 1,
        'name': 'test',
        'images': [{
            url: 'test'
        }],
        'genres': ['test', 'test']
    };

    it('renders itself without crashing', () => {
        shallow(<ArtistCard artist={artist}/>);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<ArtistCard artist={artist}/>);
        expect(tree).toMatchSnapshot();
    });
});