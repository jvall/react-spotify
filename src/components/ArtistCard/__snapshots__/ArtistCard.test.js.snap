exports[`<ArtistCard /> renders correctly 1`] = `
<div
  className="ArtistCard">
  <img
    src="test" />
  <div
    className="name">
    test
  </div>
  <div
    className="genres">
    test, test
  </div>
  <button
    onClick={[Function]}>
    Like
  </button>
</div>
`;
